### PR TITLE
Finishes Newfood repathing, kills oldfood.

### LINKED ISSUE #10391: [Bug]: Dog treats are ERRORs
### BYOND Version

514.1589

### Round Date

2023/12/24

### Round ID

47083

### Test merges

[#9124: 'Enable serverside 515.1614' by Crossedfall at commit 347bf8b17f](https://www.github.com/beestation/beestation-hornet/pull/9124)
[#10378: 'Revert "Minor refactor to machine construction + Fake Computer Toy"' by EvilDragonfiend at commit 48953205fc](https://www.github.com/beestation/beestation-hornet/pull/10378)

### Describe the bug

When arrived at Dog's Heaven, noticed that all dog treats are ERRORs
image

### What did you expect?

Dog treats should have their own sprite, instead of ERRORs

### Reproducible steps

1. Go to Dog's Heaven
2. See Dog Treats as ERRORs

### LINKED ISSUE #10677: [Bug]:  Food Cart does not hold food
### BYOND Version

Latest

### Round Date

2024/03/01

### Round ID

47874

### Test merges

_No response_

### Describe the bug

I can't put food inside the food cart

imagen

### What did you expect?

I expected food to go inside the food cart

### Reproducible steps

Get food
Use it on the cart

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/54028

## About The Pull Request

closes #10391 
closes #10677 

Fixes all instances of food not being accepted by vendors or any machine that was checking for the former typepath instead of the new one.

Fixes some signals which would not apply because they were checking for oldfood typepaths (i.e. infective component)

Replaces all remaining instance of the old food typepath with the new ones.

Removes all remaining customfood (Kebabs & Ashlander soups), all of which are to be readded in the next pr

Converts to newfood:
- snacks.dm
- snacks_other.dm
- snacks_vend.dm

Adds & adjusts protein values for all food in misc & snacks_vend

## Why It's Good For The Game

Newfood atomization

Its finally over.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Operating a processor

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/12241fd6-cdaf-4f2a-9e7d-40f6fb489f79

Baking a Birthday Cake

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/c7d39179-c90a-4bab-a36e-9cdff34186e1

</details>

## Changelog
:cl:
add: Full conversion to the new food typepath
remove: removes all customfood, TBA in the next pr
fix: fixes all instances of machines checking for the old path instead of the new food path, incl. vendors & microwaves
tweak: converts misc.dm to use protein values
tweak: converts snacks_vend.dm to use protein values
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
Meat + Plants atomizations are approaching 3 & 1 month ages respectively, so I felt it prudent to just get this through the door.

Included are the commits from those prs, I will squash and rebase them out should either be merged.

**Comment by Tsar-Salat** [comment]
Rebased to remove defunct pre-meat & pre-plants commits

**Comment by itsmeow** [comment]
Can you fix all the UpdatePaths being reversed in all the newfood scripts?

File: tools/UpdatePaths/Scripts/9951_NEWFOODmeats [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 /obj/item/food/stewedsoymeat : /obj/item/reagent_containers/food/snacks/stewedsoymeat{@OLD}
 /obj/item/food/boiledspiderleg : /obj/item/reagent_containers/food/snacks/boiledspiderleg{@OLD}
-/obj/item/food/spidereggsham : /obj/item/reagent_containers/food/snacks/spidereggsham{@OLD}
+/obj/item/food/spidereggsham : /obj/item/food/spidereggsham{@OLD}
```
Comment (itsmeow) [inline tools/UpdatePaths/Scripts/9951_NEWFOODmeats]: this is a noop?

**Comment by Tsar-Salat** [comment]
I hate my life

**Comment by Tsar-Salat** [comment]
I'll keep this open for a bit longer

**Comment by Rukofamicom** [comment]
Original PR author has deleted their account, impossible to reasonably give credit beyond linking the original PR

File: code/game/objects/items/food/snacks.dm [hunk 2]
```diff
@@ -2,145 +2,146 @@
...
-	trash = /obj/item/trash/sosjerky
-	list_reagents = list(
-		/datum/reagent/consumable/nutriment = 1,
-		/datum/reagent/consumable/sugar = 3,
+	trash_type = /obj/item/trash/sosjerky
+	food_reagents = list(
+		/datum/reagent/consumable/nutriment/protein = 3,
+		/datum/reagent/consumable/sugar = 2,
 		/datum/reagent/consumable/sodiumchloride = 2,
 		/datum/reagent/consumable/maltodextrin = 2
```
Comment (Rukofamicom) [inline code/game/objects/items/food/snacks.dm]: Maltodextrin should be updated to 3, to keep it in line with the increased nutrition of this food. 

1 Nutriment = 15 nutrition
3 Protein = 27 nutrition

**Review by Rukofamicom — CHANGES_REQUESTED** [review]

File: code/game/objects/items/food/snacks.dm (hunk 2 above)
Comment (Tsar-Salat) [inline code/game/objects/items/food/snacks.dm]:
```suggestion
		/datum/reagent/consumable/maltodextrin = 3
```

**Review by Rukofamicom — APPROVED** [review]

**Pull request merged**